src/main/java/rx/internal/util/RxRingBuffer.java

```diff
@@ -261,62 +261,34 @@
     }
     public static final int SIZE;
 
-    /* Public so Schedulers can manage the lifecycle of the inner worker. */
-    public static final ObjectPool<Queue<Object>> SPSC_POOL = new ObjectPool<Queue<Object>>() {
-
-        @Override
-        protected SpscArrayQueue<Object> createObject() {
-            return new SpscArrayQueue<Object>(SIZE);
-        }
-
-    };
-
-    /* Public so Schedulers can manage the lifecycle of the inner worker. */
-    public static final ObjectPool<Queue<Object>> SPMC_POOL = new ObjectPool<Queue<Object>>() {
-
-        @Override
-        protected SpmcArrayQueue<Object> createObject() {
-            return new SpmcArrayQueue<Object>(SIZE);
-        }
-
-    };
-
     public static RxRingBuffer getSpscInstance() {
         if (UnsafeAccess.isUnsafeAvailable()) {
-            return new RxRingBuffer(SPSC_POOL, SIZE);
+            return new RxRingBuffer(false, SIZE);
         } else {
             return new RxRingBuffer();
         }
     }
 
     public static RxRingBuffer getSpmcInstance() {
         if (UnsafeAccess.isUnsafeAvailable()) {
-            return new RxRingBuffer(SPMC_POOL, SIZE);
+            return new RxRingBuffer(true, SIZE);
         } else {
             return new RxRingBuffer();
         }
     }
 
     private RxRingBuffer(Queue<Object> queue, int size) {
         this.queue = queue;
-        this.pool = null;
         this.size = size;
     }
 
-    private RxRingBuffer(ObjectPool<Queue<Object>> pool, int size) {
-        this.pool = pool;
-        this.queue = pool.borrowObject();
+    private RxRingBuffer(boolean spmc, int size) {
+        this.queue = spmc ? new SpmcArrayQueue<Object>(size) : new SpscArrayQueue<Object>(size);
         this.size = size;
     }
 
     public synchronized void release() { // NOPMD
```

Comment:
Thought about it but it's a semi-public API and I didn't want to break it just that.